### PR TITLE
BUG: to_numeric downcast = 'unsigned' would not un-sign a 0 value.

### LINKED ISSUE #14401: BUG: pd.to_numeric(..., downcast='unsigned') should accept 0
with 0.19.0.

```
In [28]: pd.to_numeric(pd.Series([1,2,3]), downcast='unsigned')
Out[28]: 
0    1
1    2
2    3
dtype: uint8

In [29]: pd.to_numeric(pd.Series([0,1,2,3]), downcast='unsigned')
Out[29]: 
0    0
1    1
2    2
3    3
dtype: int64
```

```
In [30]: np.iinfo(np.uint8).min
Out[30]: 0
```

I think 0 is a legit uint8, so [29] should be uint8.

### PR DESCRIPTION
**Pull request by verhalenn**
- [ ] closes #14401
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

Have to figure out whats going wrong with the build still. Will let you know as soon as I do.

### CODE REVIEW

File: doc/source/whatsnew/v0.19.1.txt [hunk 1]
```diff
@@ -78,3 +78,7 @@ Bug Fixes
 
 
 - Bug in ``pd.pivot_table`` may raise ``TypeError`` or ``ValueError`` when ``index`` or ``columns`` is not scalar and ``values`` is not specified (:issue:`14380`)
+- Bug in ``pd.pivot_table`` may raise ``TypeError`` or ``ValueError`` when ``index`` or ``columns``
+- Bug in ``pd.to_numeric`` where a 0 was not unsigned on a downcast = 'unsigned' argument (:issue:`14401`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.1.txt]: just add your fix, don't change the existing

**Comment by jreback** [comment]
can you rebase this

**Comment by jorisvandenbossche** [comment]
@verhalenn There is still a test failing. Do you have time to look into this?

**Comment by verhalenn** [comment]
@jorisvandenbossche I have been but I'm having trouble replicating it.

**Comment by jorisvandenbossche** [comment]
@verhalenn Apparently this is a bug in older numpy versions:

```
>>> np.__version__
'1.8.2'
>>> arr = np.array([-128, 127])
>>> arr2 = arr.astype('int8')
>>> np.allclose(arr, arr2)
False
```

```
>>> np.__version__
'1.11.2'
>>> arr = np.array([-128, 127])
>>> arr2 = arr.astype('int8')
>>> np.allclose(arr, arr2)
True
```

So you can skip the failing test for numpy versions of 1.8 or lower.

**Comment by verhalenn** [comment]
That should do. If there is anything you would like different, let me know.

**Comment by jorisvandenbossche** [comment]
@verhalenn See here for an example how to skip a test for a numpy version: https://github.com/pandas-dev/pandas/blob/52f31d470d779204e4c1388cdb56351c68332c3f/pandas/tests/series/test_quantile.py#L76

Can you also move the whatsnew note to the 0.19.2 file ?

File: doc/source/whatsnew/v0.19.1.txt [hunk 2]
```diff
@@ -58,4 +58,4 @@ Bug Fixes
 - Bug in ``df.groupby`` causing an ``AttributeError`` when grouping a single index frame by a column and the index level (:issue`14327`)
 - Bug in ``df.groupby`` where ``TypeError`` raised when ``pd.Grouper(key=...)`` is passed in a list (:issue:`14334`)
 - Bug in ``pd.pivot_table`` may raise ``TypeError`` or ``ValueError`` when ``index`` or ``columns``
-  is not scalar and ``values`` is not specified (:issue:`14380`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.1.txt]: move to 0.19.2

File: doc/source/whatsnew/v0.19.2.txt [hunk 3]
```diff
@@ -25,3 +25,4 @@ Bug Fixes
 
 - compat with ``dateutil==2.6.0`` for testing (:issue:`14621`)
 - allow ``nanoseconds`` in ``Timestamp.replace`` kwargs (:issue:`14621`)
+- Bug in ``pd.to_numeric`` where a 0 was not unsigned on a downcast = 'unsigned' argument (:issue:`14401`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.2.txt]: ``downcast='unsigned'`` (use double-backticks)

File: pandas/tools/tests/test_util.py [hunk 4]
```diff
@@ -401,6 +403,39 @@ def test_downcast(self):
...
+        u = 'unsigned'
+        dtype_downcast_min_max = [
+            ('int8', i, [iinfo(int8).min, iinfo(int8).max]),
+            ('int16', i, [iinfo(int16).min, iinfo(int16).max]),
+            ('int32', i, [iinfo(int32).min, iinfo(int32).max]),
+            ('int64', i, [iinfo(int64).min, iinfo(int64).max]),
+            ('uint8', u, [iinfo(uint8).min, iinfo(uint8).max]),
+            ('uint16', u, [iinfo(uint16).min, iinfo(uint16).max]),
+            ('uint32', u, [iinfo(uint32).min, iinfo(uint32).max]),
+            # ('uint64', u, [iinfo(uint64).min, iinfo(uint64).max]),
```
Comment (jreback) [inline pandas/tools/tests/test_util.py]: put a comment on why skipping the ``uint64``

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**